pkg/login/ldap_login_test.go

```diff
@@ -8,65 +8,41 @@ import (
 	"github.com/grafana/grafana/pkg/services/ldap"
 	"github.com/grafana/grafana/pkg/services/multildap"
 	"github.com/grafana/grafana/pkg/setting"
-	. "github.com/smartystreets/goconvey/convey"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
 )
 
 var errTest = errors.New("test error")
 
-func TestLDAPLogin(t *testing.T) {
-	Convey("Login using ldap", t, func() {
-		Convey("Given ldap enabled and no server configured", func() {
-			setting.LDAPEnabled = true
+func TestLoginUsingLDAP(t *testing.T) {
+	LDAPLoginScenario(t, "When LDAP enabled and no server configured", func(sc *LDAPLoginScenarioContext) {
+		setting.LDAPEnabled = true
 
-			LDAPLoginScenario("When login", func(sc *LDAPLoginScenarioContext) {
-				sc.withLoginResult(false)
-				getLDAPConfig = func(*setting.Cfg) (*ldap.Config, error) {
-					config := &ldap.Config{
-						Servers: []*ldap.ServerConfig{},
-					}
-
-					return config, nil
-				}
-
-				enabled, err := loginUsingLDAP(sc.loginUserQuery)
-
-				Convey("it should return true", func() {
-					So(enabled, ShouldBeTrue)
-				})
-
-				Convey("it should return no LDAP servers error", func() {
-					So(err, ShouldEqual, errTest)
-				})
+		sc.withLoginResult(false)
+		getLDAPConfig = func(*setting.Cfg) (*ldap.Config, error) {
+			config := &ldap.Config{
+				Servers: []*ldap.ServerConfig{},
+			}
 
-				Convey("it should not call ldap login", func() {
-					So(sc.LDAPAuthenticatorMock.loginCalled, ShouldBeTrue)
-				})
-			})
-		})
+			return config, nil
+		}
 
-		Convey("Given ldap disabled", func() {
-			setting.LDAPEnabled = false
+		enabled, err := loginUsingLDAP(sc.loginUserQuery)
+		require.EqualError(t, err, errTest.Error())
 
-			LDAPLoginScenario("When login", func(sc *LDAPLoginScenarioContext) {
-				sc.withLoginResult(false)
-				enabled, err := loginUsingLDAP(&models.LoginUserQuery{
-					Username: "user",
-					Password: "pwd",
-				})
+		assert.True(t, enabled)
+		assert.True(t, sc.LDAPAuthenticatorMock.loginCalled)
+	})
 
-				Convey("it should return false", func() {
-					So(enabled, ShouldBeFalse)
-				})
+	LDAPLoginScenario(t, "When LDAP disabled", func(sc *LDAPLoginScenarioContext) {
+		setting.LDAPEnabled = false
```

Comment:
Not entirely within the scope of the PR, but we should fix this while we're cleaning up:
```suggestion
		origLDAPEnabled = setting.LDAPEnabled
		t.Cleanup(func() {
			setting.LDAPEnabled = origLDAPEnabled
		})
		setting.LDAPEnabled = true
```